src/agent/sqlite_manager/src/sqlite_manager.cpp

```diff
@@ -0,0 +1,311 @@
+    ColumnType SQLiteManager::ColumnTypeFromSQLiteType(const int type) const
+    {
+        if (type == SQLite::INTEGER)
+            return ColumnType::INTEGER;
+
+        if (type == SQLite::FLOAT)
+            return ColumnType::FLOAT;
+
+        return ColumnType::TEXT;
+    }
+
+    SQLiteManager::SQLiteManager(const std::string& dbName)
+        : m_dbName(dbName)
+        , m_db(std::make_unique<SQLite::Database>(dbName, SQLite::OPEN_READWRITE | SQLite::OPEN_CREATE))
+    {
+        // Open the database in WAL mode
+        m_db->exec("PRAGMA journal_mode=WAL;");
+    }
+
+    void SQLiteManager::CreateTable(const std::string& tableName, const std::vector<Col>& cols)
+    {
+        std::vector<std::string> fields;
+        for (const Col& col : cols)
+        {
+            std::string field = fmt::format("{} {}{}{}{}",
+                                            col.m_name,
+                                            mapColTypeString.at(col.m_type),
+                                            (col.m_primaryKey) ? " PRIMARY KEY" : "",
+                                            (col.m_autoIncrement) ? " AUTOINCREMENT" : "",
+                                            (col.m_notNull) ? " NOT NULL" : "");
+            fields.push_back(field);
+        }
+
+        std::string queryString =
+            fmt::format("CREATE TABLE IF NOT EXISTS {} ({})", tableName, fmt::format("{}", fmt::join(fields, ", ")));
+
+        std::lock_guard<std::mutex> lock(m_mutex);
+        try
+        {
+            m_db->exec(queryString);
+        }
+        catch (const std::exception& e)
+        {
+            std::cerr << "Error initializing table: " << e.what() << std::endl;
+            throw;
+        }
+    }
+
+    void SQLiteManager::Insert(const std::string& tableName, const std::vector<Col>& cols)
+    {
+        std::vector<std::string> names;
+        std::vector<std::string> values;
+
+        for (const Col& col : cols)
+        {
+            names.push_back(col.m_name);
+            if (col.m_type == ColumnType::TEXT)
+            {
+                values.push_back(fmt::format("'{}'", col.m_value));
+            }
+            else
+                values.push_back(col.m_value);
+        }
+
+        std::string queryString =
+            format("INSERT INTO {} ({}) VALUES ({})", tableName, fmt::join(names, ", "), fmt::join(values, ", "));
+
+        std::lock_guard<std::mutex> lock(m_mutex);
+        try
+        {
+            m_db->exec(queryString);
+        }
+        catch (const std::exception& e)
+        {
+            std::cerr << "Error inserting element: " << e.what() << std::endl;
+            throw;
+        }
+    }
+
+    int SQLiteManager::GetCount(const std::string& tableName)
+    {
+        std::string queryString = fmt::format("SELECT COUNT(*) FROM {}", tableName);
+
+        try
+        {
+            SQLite::Statement query(*m_db, queryString);
+            int count = 0;
+            if (query.executeStep())
```

Comment:
Missing mutex lock. Move the execution of the query to a private function and use it in all the functions that need it.

Reply:
Added mutex lock.
Moved query execution code to its own function for most db operations. In those that require handling results, such as this particular case, it is not possible.